Sources/TuistCore/Graph/XCFrameworkNode.swift

```diff
@@ -0,0 +1,65 @@
+import Basic
+import Foundation
+import TuistSupport
+
+final class XCFrameworkNode: PrecompiledNode {
+    let libraries: [XCFrameworkInfoPlist.Library]
+
+    init(path: AbsolutePath, libraries: [XCFrameworkInfoPlist.Library]) {
+        self.libraries = libraries
+        super.init(path: path)
+    }
+
+    static func parse(
+        path: AbsolutePath,
+        cache: GraphLoaderCaching,
+        configurationFile: String = "Info.plist",
+        fileHandler: FileHandling = FileHandler.shared
+    ) throws -> XCFrameworkNode {
+        if let xcframeworkNode = cache.precompiledNode(path) as? XCFrameworkNode {
+            return xcframeworkNode
+        }
+        
+        let infoPlist = path.appending(component: configurationFile)
```

Comment:
As I mentioned in my comment, I would not introduce side effects in the constructors of the models. It complicates tests and makes it hard to introduce future optimizations.